### ISSUE TITLE
Cannot add nonce using default ASP .NET Core middleware

### ISSUE THREAD

**Issue by asgerl** [MiniProfiler/dotnet#547]
I have setup an ASP .NET Core application with _MiniProfiler_.

I wish to profile certain parts of the application and display the profilings in the MiniProfiler UI.

To reduce maintenance I have decided to use the default `MiniProfilerMiddleware` - instead of creating my own middleware.

The `MiniProfilerMiddleware` both profiles my application and renders the _MiniProfiler UI_.

Recently I have added a _Content Security Policy_ to my application. As recommended I have disabled `UnsafeInlineSrc`. This however conflicts with _MiniProfiler UI_ which uses inline JavaScript - meaning I can no longer render _MiniProfiler UI_.

For now I have enabled `UnsafeInlineSrc` - against _Content Security Policy_ recommendations. My preferred solution would be to use nonce for the MiniProfiler UI.

I can see that I am not the first experiencing this issue with the _MiniProfiler UI_ - in 4.2.1 [nonce supports](https://github.com/MiniProfiler/dotnet/pull/465) was added to make _MiniProfiler UI_ more CSP compliant.

However I cannot find a way to add a nonce when using the `MiniProfilerMiddleware`. E.g. [MiniProfilerOptions.cs](https://github.com/MiniProfiler/dotnet/blob/main/src/MiniProfiler.AspNetCore/MiniProfilerOptions.cs) does not expose Nonce as an option.

Also from inspecting the code it seems that the new [RenderOptions](https://github.com/MiniProfiler/dotnet/pull/451) are not used at all for rendering the MiniProfiler pages generated by the [MiniProfilerMiddleware](https://github.com/MiniProfiler/dotnet/blob/b5907860ef62c2c4de51a73b1d539c606cefe5ce/src/MiniProfiler.Shared/Internal/Render.cs#L261).

It seems currently I cannot add a nonce to MiniProfiler UI and must instead look for alternative solutions, e.g.:
- Create my own `MiniProfilerMiddleware`.
- Keep having enabled `UnsafeInlineSrc` enabled.

Am I missing something? is there a way to add a nonce to the UI when using the default MiniProfilerMiddlware? Is this something that will be supported at some point?

**Comment by NickCraver** [comment]
Where is your `nonce` coming from? It can be set in the UI, yep - for example `<mini-profiler nonce="@MyNonce()"/>` in your view where it's rendering the script.

**Comment by pkunze** [comment]
The Problem isn't the TagHelper if thats what you meant? 
We are facing the same Problem in results index.

It doesn't load a script in the results-index-page:

```
Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'sha256-//zTN8XhURvVZwifRny+gOFsLioyADgO9Nx3svK7jaU=' 'nonce-aC+A6rfoE0X3D1IF2aRVnWJrYF5QYQeLOy+m0xPhCR0='". Either the 'unsafe-inline' keyword, a hash ('sha256-Ms0C0ZmShuW22uPfBLUX7PsrkLcBCrRvQKP4HqylLYA='), or a nonce ('nonce-...') is required to enable inline execution.
```

We are suspecting this part:

![image](https://user-images.githubusercontent.com/1416798/149891414-798c9aa2-2d36-4e36-98fe-a6d49dc4dc94.png)

**Comment by pkunze** [comment]
I think the solution would be a way to add a request-specific `nonce`-Attribute as described by @asgerl.
We would gladly investigate it and file a PR.

**Comment by NickCraver** [comment]
Working on #634 and making sure to add support for all these cases. I think this will be a usable model for everything said here (and I'm adding it as a demo in the samples project).

**Comment by NickCraver** [comment]
Now available with #634 with samples in the AspNet application (I'm not sure how others are plugging in the nonce given there doesn't seem to be an official way yet, but did a simple custom service there).
